bc_obps/reporting/service/compliance_service.py

```diff
@@ -288,3 +294,45 @@ def get_calculated_compliance_data(cls, report_version_id: int) -> ComplianceDat
         )
 
         return return_object
+
+    @classmethod
+    @transaction.atomic()
+    def save_compliance_data(cls, report_version_id: int) -> None:
+        compliance_data_to_save = ComplianceService.get_calculated_compliance_data(report_version_id)
+        compliance_summary_record_id = None
+        report_version_record = ReportVersion.objects.get(id=report_version_id)
+        if ReportComplianceSummary.objects.filter(report_version_id=report_version_id).exists():
+            compliance_summary_record_id = ReportComplianceSummary.objects.get(report_version_id=report_version_id).id
+        compliance_summary_record, _ = ReportComplianceSummary.objects.update_or_create(
+            id=compliance_summary_record_id,
+            report_version=report_version_record,
+            emissions_attributable_for_reporting=compliance_data_to_save.emissions_attributable_for_reporting,
+            reporting_only_emissions=compliance_data_to_save.reporting_only_emissions,
+            emissions_attributable_for_compliance=compliance_data_to_save.emissions_attributable_for_compliance,
+            emissions_limit=compliance_data_to_save.emissions_limit,
+            excess_emissions=compliance_data_to_save.excess_emissions,
+            credited_emissions=compliance_data_to_save.credited_emissions,
+            reduction_factor=compliance_data_to_save.regulatory_values.reduction_factor,
+            tightening_rate=compliance_data_to_save.regulatory_values.tightening_rate,
+            initial_compliance_period=compliance_data_to_save.regulatory_values.initial_compliance_period,
+            compliance_period=compliance_data_to_save.regulatory_values.compliance_period,
+        )
```

Comment:
`update_or_create` works a bit differently and will try to match every named field we pass, to apply `defaults` or `create_defaults` to the matching records, depending on the scenario: https://docs.djangoproject.com/en/5.2/ref/models/querysets/#update-or-create

This should be (with the dataclass usage it becomes very easy):
```py
compliance_summary_record, _ = ReportComplianceSummary.objects.update_or_create(
    id=compliance_summary_record_id,
    report_version=report_version_record,
    defaults=dataclasses.asdict(compliance_data_to_save)
)
```